source/base/partitioner.cc

```diff
@@ -199,7 +199,7 @@ namespace Utilities
       if (my_pid == 0)
         my_size += local_range_data.first;
 
-      types::global_dof_index my_shift = 0;
+      types::global_dof_index my_shift = local_range_data.first;
```

Comment:
this is to allow gaps between processors, right? please leave a comment.

Reply:
@kronbichler Could you explain how I should read this line? Why is the value `0` not correct? It is overridden in the following lines, isn't it? 

```cpp
      types::global_dof_index my_shift = local_range_data.first;
      {
        const int ierr = MPI_Exscan(&my_size,
                                    &my_shift,
                                    1,
                                    DEAL_II_DOF_INDEX_MPI_TYPE,
                                    MPI_SUM,
                                    communicator);
        AssertThrowMPI(ierr);
      }
```

Reply:
`MPI_Exscan` computes the sum from the processor 0 up to (excluding) the current rank. Thus, on rank 0, we lack the knowledge on the shift.

Reply:
makes sense. @peterrum can you leave a comment that states this?

Reply:
I have slightly rewritten the code. That MPI does not alter the value of rank 0 is as far I see it not documented. I would have expected it is set to zero, since it an output (and is the logical result of an exclusive scan). However, [this side](https://mpi.deino.net/mpi_functions/MPI_Exscan.html) states:

> The value in recvbuf on the process with rank 0 is undefined, and recvbuf is not signficant on process 0.

To be safe, I set the value on rank 0 **after** `MPI_Exscan`.

ping @tjhei and @kronbichler